app/controllers/form_question_controller.rb

```diff
@@ -3,28 +3,79 @@ class FormQuestionController < ApplicationController
   def show
     @form_type = params[:type]
     @list_of_questions = FormQuestion.get_questions_for_form(@form_type)
-  	if (params[:submit]) 
-      if form_completed?
-        flash[:notice] = "#{@form_type} successfully submitted."
-      	redirect_to user_path()
-      else
-      	@error = "Please complete the form before submitting."
+    application = User.find(params[:id]).get_most_recent_application
+    if application
+      @saved_form = application.content
+      if @saved_form.has_key?(@form_type)
+        @form_answer = get_answers_to_prefill_from(@saved_form[@form_type])
+      end
+    end
+  end
+
+  def save_progress
+    @user = User.find(params[:id])
+    @form_content = get_form_content
+    @application = @user.get_most_recent_application
+    if params[:commit] == "Save and Return"
+      update_application(false)
+      flash[:notice] = "#{params[:type]} successfully saved."
+      redirect_to user_path(@user)
+    elsif params[:commit] == 'Submit'
+      submit
+    end
+  end
+
```

Comment:
The rest of these methods should be private to show they are not part of the routes.